### PR TITLE
chore: deploy manifest for CLI and VSCode

### PR DESCRIPTION
**Pull request by 1yefuwang1**

### CODE REVIEW

File: packages/fx-core/src/common/constants.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export class FeatureFlagName {
   static readonly YoCheckerEnable = "TEAMSFX_YO_ENV_CHECKER_ENABLE";
   static readonly GeneratorCheckerEnable = "TEAMSFX_GENERATOR_ENV_CHECKER_ENABLE";
   static readonly ApiConnect = "TEAMSFX_API_CONNECT_ENABLE";
+  static readonly DeployManifest = "TEAMSFX_DEPLOY_MANIFEST";
```
Comment (tecton) [inline packages/fx-core/src/common/constants.ts:78]: Could you please help to add this to `initializeGAFeatureFlags()` as well?

Comment (1yefuwang1) [reply packages/fx-core/src/common/constants.ts:78]: Sure, added. Thx for reminding.

File: packages/cli/src/cmds/index.ts [hunk 2]
```diff
@@ -41,6 +42,10 @@ export const commands: YargsCommand[] = [
   new Env(),
 ];
 
+if (isDeployManifestEnabled()) {
+  commands.splice(8, 1);
```
[the remark this answers is not in the document]
Comment (a1exwang) [reply packages/cli/src/cmds/index.ts:46]: Also, when `isGAPreviewEnabled() == false`, is the logic correct?

**Review by tecton — DISMISSED** [review]

**Review by nliu-ms — APPROVED** [review]

File: packages/cli/src/cmds/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (1yefuwang1) [reply packages/cli/src/cmds/index.ts]: Good point. Changed. Thx.

**Review by a1exwang — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]
LGTM

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**